packages/react/src/useWalletAccountTransactionSendingSigner.ts

```diff
@@ -0,0 +1,54 @@
+import { address } from '@solana/addresses';
+import { SOLANA_ERROR__SIGNER__WALLET_MULTISIGN_UNIMPLEMENTED, SolanaError } from '@solana/errors';
+import { SignatureBytes } from '@solana/keys';
+import { TransactionSendingSigner } from '@solana/signers';
+import { getTransactionEncoder, Transaction } from '@solana/transactions';
+import { UiWalletAccount } from '@wallet-standard/ui';
+import { useMemo, useRef } from 'react';
+
+import { getAbortablePromise } from './abortable-promise';
+import { OnlySolanaChains } from './chain';
+import { useSignAndSendTransaction } from './useSignAndSendTransaction';
+
+/**
+ * Returns an object that conforms to the `TransactionSendingSigner` interface of `@solana/signers`.
+ */
+export function useWalletAccountTransactionSendingSigner<TWalletAccount extends UiWalletAccount>(
+    uiWalletAccount: TWalletAccount,
+    chain: OnlySolanaChains<TWalletAccount['chains']>,
+    extraConfig?: Readonly<{
+        getOptions(transaction: Transaction): Readonly<{ minContextSlot?: bigint }> | undefined;
+    }>,
```

Comment:
Ugh. This is actually really awkward. This is not really a ‘creating a signer’ level option; it's a per-transaction-being-signed option. There's no way in the `@solana/signers` API to be like ‘hey sign this transaction using these signers but with these weird `options` mixed in.’

I updated the API so that you can pass in a function to produce the options, on demand, given the particular transaction in question.